pkg/os/copy.go

```diff
@@ -64,12 +69,13 @@ func CopySparse(dst io.WriteSeeker, src io.Reader) (int64, error) {
 }
 
 type sparseWriter struct {
+	context         context.Context
```

Comment:
I don't know how much this would complexify this code, but https://pkg.go.dev/context#Background says:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. 

Looking some more into it, I have no idea how to combine this recommandation with a cancellable `Write`

Reply:
`exec.CommandContext` and `http.Request` do not follow this guideline though, and store a `ctx` in a struct, let's just ignore my comment.